src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1606,10 +1606,7 @@ protected internal override void OnProjectRemoved(ProjectId projectId)
                 base.OnProjectRemoved(projectId);
 
                 _threadingContext.RunWithShutdownBlockAsync(async cancellationToken =>
-                {
-                    await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
-                    RefreshProjectExistsUIContextForLanguage(languageName);
-                });
+                    await RefreshProjectExistsUIContextForLanguageAsync(languageName, cancellationToken).ConfigureAwait(false));
```

Comment:
Make the lambda no longer async? It'll then remove the ConfigureAwait(false) which is only a bit funky here since we're potentially putting in an extra jump back to the thread pool...maybe.

Reply:
done.